stdlib/typing.pyi

```diff
@@ -797,11 +797,15 @@ class MutableMapping(Mapping[_KT, _VT]):
     # -- weakref.WeakValueDictionary.__ior__
     # -- weakref.WeakKeyDictionary.__ior__
     @overload
-    def update(self, m: SupportsKeysAndGetItem[_KT, _VT], /, **kwargs: _VT) -> None: ...
+    def update(self, m: SupportsKeysAndGetItem[_KT, _VT], /) -> None: ...
     @overload
-    def update(self, m: Iterable[tuple[_KT, _VT]], /, **kwargs: _VT) -> None: ...
+    def update(self: Mapping[str, _VT], m: SupportsKeysAndGetItem[_KT, _VT], /, **kwargs: _VT) -> None: ...
```

Comment:
Should this be:

```python
def update(self: Mapping[str, _VT], m: SupportsKeysAndGetItem[str, _VT], /, **kwargs: _VT) -> None: ...
```

It's possible that `_KT` is constrained to `str` anyway via the self mapping, but I still think being explicit would be better (and easier on the type checkers).

Reply:
I think SupportsKeysAndGetItem is invariant in the first arg so that might cause trouble?

Reply:
#14100 for these cleanups.